### ISSUE TITLE
Problem with tproxy in docker container when different iptables are present on host and container

### ISSUE THREAD

**Issue by bartsmykla** [kumahq/kuma#6334]
## Initial data necessary to better understand the problem
- versions of `iptables` binary inside images
  - `iptables v1.8.7 (legacy)` for `kumahq/kuma-init:2.1.1`
  - `iptables v1.8.7 (nf_tables)` for `kumahq/kuma-universal:0.0.0-preview.vlocal-build` locally built for universal e2e tests
- > Docker containers that attach to a [custom network](https://docs.docker.com/network/network-tutorial-standalone/#use-user-defined-bridge-networks) use Docker’s embedded DNS server. The embedded DNS server forwards external DNS lookups to the DNS servers configured on the host.
  > ref: https://docs.docker.com/config/containers/container-networking/#dns-services
  - docker achieves this by applying [iptables rules](https://github.com/docker/engine/blob/master/libnetwork/resolver_unix.go) inside [container's network namespace](https://github.com/docker/engine/blob/master/libnetwork/resolver_unix.go#L55-L59)
- to simplify thinking about the problem, let's assume we create a mental abstraction called `bucket`. A `bucket` can contain iptables rules.
  - when using `iptables-legacy`, rules land in `[iptables]` bucket
  - when using `iptables-nft`, rules land in `[nftables]` bucket

## Problem

If we want/have to use docker's embedded DNS resolver, we have to install transparent proxy with a flag `--redirect-dns-upstream-target-chain="DOCKER_OUTPUT"`.
If there is discrepancy between versions of `iptables` on the host and container, the installation might fail with similar error:
```sh
Error: failed to setup transparent proxy: cannot restore ipv4 iptable rules: executing command failed: exit status 2 (with output: "iptables-restore v1.8.7 (nf_tables): Chain 'DOCKER_OUTPUT' does not exist
Error occurred at line: 21
Try `iptables-restore -h' or 'iptables-restore --help' for more information.")
```

## Why

As described in the first section. To allow container to send DNS queries to Docker's embedded DNS resolver, Docker uses host's `iptables` and adds appropriate rules to container's network namespace.

### Example:
- If host uses `iptables-legacy`, the embedded dns resolver rules mentioned earlier will land in container's `[iptables]` bucket
- If container comes with `iptables-nft`, after installing transparent proxy, rules will land in container's `[nftables]` rules

As to install transparent proxy inside a container we expect `DOCKER_OUTPUT` chain to exist, and as the iptables rule to create this chain landed in different "bucket", we don't have access to it, and thus the whole installation fails.

**Comment by jakubdyszkiewicz** [comment]
> Triage: when `--redirect-dns-upstream-target-chain="DOCKER_OUTPUT"` is defined. Try to first apply using iptables-nft. If fails, use iptables-legacy.

**Comment by lahabana** [comment]
@bartsmykla is this solved by you working on #8293?

**Comment by lahabana** [comment]
Fixed by: https://github.com/kumahq/kuma/pull/9750
